### ISSUE TITLE
Error in rose-task-hook2

### ISSUE THREAD

**Issue by penguian** [ACCESS-NRI/accessdev-Trac-archive#142]
`resolution_fixed`   |    by mrd599@nci.org.au

___

When rose-task-hook2 is used with cylc handlers with compound names, e.g.
```
execution timeout handler = "rose task-hook2 --mail"
```
rose gives an error
```
  File "/usr/local/rose/lib/python/rose/suite_hook.py", line 124, in main
    args = suite_engine_proc.process_suite_hook_args(*args, **vars(opts))
  File "/usr/local/rose/lib/python/rose/suite_engine_procs/cylc.py", line 954, in process_suite_hook_args
    hook_event, suite, task, hook_message = args
ValueError: too many values to unpack
          execution timeout handler = "rose task-hook2 --mail"
```

Adding a diagnostic print shows it's being called with args
```
('execution', 'timeout', 'au_aa120_orig', 'atmos_no_omp_no_prebuild.1', 'job started 1.0 minutes ago, but has not finished')
```

whereas suite-hook gives
```
('execution timeout', 'au_aa120_orig', 'atmos_no_omp_no_prebuild.1', 'job started 1.0 minutes ago, but has not finished')
```

rose-task-hook2 has
```
   rose task-hook $opt $STATE $SUITE $JOB "$MSG" 
```  
but this should be
```
   rose task-hook $opt "$STATE" $SUITE $JOB "$MSG"
```

___
_Issue migrated from [trac:142](https://accessdev.nci.org.au/trac/142) at 2024-01-31 18:07:45 +1100_

**Comment by penguian** [comment]
_Scott Wales <scott.wales@unimelb.edu.au>_ _changed status from `new` to `closed`_

**Comment by penguian** [comment]
_Scott Wales <scott.wales@unimelb.edu.au>_ _changed owner from `` to `Scott Wales <scott.wales@unimelb.edu.au>`_

**Comment by penguian** [comment]
_Scott Wales <scott.wales@unimelb.edu.au>_ _changed resolution from `` to `fixed`_

**Comment by penguian** [comment]
_Scott Wales <scott.wales@unimelb.edu.au>_ _committed [16425]_
___
In [changeset:"16425fb905cd4b2846aae3a26811d9257165716f/puppet"]:
```
#CommitTicketReference repository="puppet" revision="16425fb905cd4b2846aae3a26811d9257165716f"
Quote variables in task-hook2

The hook is failing when the 'execution timeout' hook is called, due to
the space in the name. Quote variables to keep them together

Fixes #142
```

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _changed status from `closed` to `reopened`_

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _changed resolution from `fixed` to ``_

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _commented_


___
Now failing with a different error. 
```
+ rose task-hook ' -v -v' succeeded au-aa123 fcm_make.2014072400 'job succeeded'
Traceback (most recent call last):
  File "/usr/lib64/python2.6/runpy.py", line 122, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.6/runpy.py", line 34, in _run_code
    exec code in run_globals
  File "/usr/local/rose/lib/python/rose/suite_hook.py", line 135, in <module>
    main()
  File "/usr/local/rose/lib/python/rose/suite_hook.py", line 124, in main
    args = suite_engine_proc.process_suite_hook_args(*args, **vars(opts))
  File "/usr/local/rose/lib/python/rose/suite_engine_procs/cylc.py", line 954, in process_suite_hook_args
    hook_event, suite, task, hook_message = args
ValueError: too many values to unpack
```

The way rose-task-hook constructs the opt variable gives it a leading space. When this is quoted the space prevents the python OptionParser picking it up as an option and it gets treated as an extra unexpected argument.

rose-task-hook should use 
```
rose task-hook $opt "$STATE" "$SUITE" "$JOB" "$MSG"
```
rather than
```
rose task-hook "$opt" "$STATE" "$SUITE" "$JOB" "$MSG"
```

**Comment by penguian** [comment]
_Scott Wales <scott.wales@unimelb.edu.au>_ _changed status from `reopened` to `closed`_

**Comment by penguian** [comment]
_Scott Wales <scott.wales@unimelb.edu.au>_ _changed resolution from `` to `fixed`_

**Comment by penguian** [comment]
_Scott Wales <scott.wales@unimelb.edu.au>_ _commented_


___
In [changeset:"b65d54f9d4af5d68c297093e75a0c1379f0fcb88/puppet"]:
```
#CommitTicketReference repository="puppet" revision="b65d54f9d4af5d68c297093e75a0c1379f0fcb88"
Fix 'opt' quoting

This is a list, not a single value

Fixes #142
```

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _changed status from `closed` to `reopened`_

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _changed resolution from `fixed` to ``_

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _commented_


___
Script currently has
```
chmod a+r "$HOME/cylc-run/$SUITE/log/job/$JOB*"
```
Quoting stops the shell filename expansion and so the chmod fails. log/suite/err has messages like
```
chmod a+r '/home/599/mrd599/cylc-run/au-aa123/log/job/gcom_test.2014081800*'
chmod: cannot access `/home/599/mrd599/cylc-run/au-aa123/log/job/gcom_test.2014081800*': No such file or directory
```

**Comment by penguian** [comment]
_@robin.bowen@bom.gov.au_ _commented_


___
Suggested fix is to move the * outside of the quotes.

**Comment by penguian** [comment]
_stn599@nci.org.au_ _changed status from `reopened` to `closed`_

**Comment by penguian** [comment]
_stn599@nci.org.au_ _changed resolution from `` to `fixed`_

**Comment by penguian** [comment]
_stn599@nci.org.au_ _commented_


___
Fixed in puppet commit bf6430520a1c9aa40d8d2ec4db7b775e8ee34eb6
